### ISSUE TITLE
[Bug] streaming read with multi consumer may lead to exception on commit

### ISSUE THREAD

**Issue by dragonkid** [apache/incubator-paimon#2072]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.5.0-incubating

### Compute Engine

Flink

### Minimal reproduce step

two streaming read job with different consumer id on same table may reproduce this problem accidentally.

### What doesn't meet your expectations?

should not fail on this scene

### Anything else?

Exceptions:

```
2023-09-26 17:48:23
java.io.UncheckedIOException: org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0]
	at org.apache.paimon.utils.JsonSerdeUtil.fromJson(JsonSerdeUtil.java:59)
	at org.apache.paimon.consumer.Consumer.fromJson(Consumer.java:54)
	at org.apache.paimon.consumer.Consumer.fromPath(Consumer.java:64)
	at org.apache.paimon.consumer.ConsumerManager.consumer(ConsumerManager.java:58)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.LongPipeline.reduce(Unknown Source)
	at org.apache.paimon.consumer.ConsumerManager.minNextSnapshot(ConsumerManager.java:78)
	at org.apache.paimon.operation.FileStoreExpireImpl.expireUntil(FileStoreExpireImpl.java:128)
	at org.apache.paimon.operation.FileStoreExpireImpl.expire(FileStoreExpireImpl.java:117)
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:205)
	at org.apache.paimon.table.sink.TableCommitImpl.commitMultiple(TableCommitImpl.java:148)
	at org.apache.paimon.flink.sink.StoreCommitter.commit(StoreCommitter.java:71)
	at org.apache.paimon.flink.sink.CommitterOperator.commitUpToCheckpoint(CommitterOperator.java:157)
	at org.apache.paimon.flink.sink.CommitterOperator.notifyCheckpointComplete(CommitterOperator.java:151)
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:104)
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.notifyCheckpointComplete(RegularOperatorChain.java:145)
	at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpoint(SubtaskCheckpointCoordinatorImpl.java:467)
	at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:400)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1430)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$16(StreamTask.java:1371)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$19(StreamTask.java:1410)
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50)
	at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMail(MailboxProcessor.java:398)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMailsWhenDefaultActionUnavailable(MailboxProcessor.java:367)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:352)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:229)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:839)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:788)
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:952)
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:931)
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:745)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:562)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: UNKNOWN; line: 1]
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1746)
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.ObjectReader._initForReading(ObjectReader.java:360)
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2095)
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1546)
	at org.apache.paimon.shade.jackson2.com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1566)
	at org.apache.paimon.utils.JsonSerdeUtil.fromJson(JsonSerdeUtil.java:57)
	... 42 more
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by liming30** [comment]
@dragonkid Thanks for your reporting. I think the problem is caused by `resetConsumer` not being atomic.  This results in the consumer file being potentially empty when read.

**Comment by liming30** [comment]
Fixed in #2092
